nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-framework-core/src/main/java/org/apache/nifi/controller/repository/StandardProcessSession.java

```diff
@@ -2527,6 +2554,43 @@ public String toString() {
         return "StandardProcessSession[id=" + sessionId + "]";
     }
 
+    @Override
+    public String getUnacknowledgedFlowfileInfo() {
```

Comment:
I would make this private and have it called by rollback itself for the sole purpose of writing to the log.  If you want to return something to a client/caller then it should definitely offer some sort of object model (set of flowfiles) type return rather than a preformatted string.